java-manta-client/src/main/java/com/joyent/manta/config/BaseChainedConfigContext.java

```diff
@@ -312,7 +312,7 @@ public String getEncryptionPrivateKeyPath() {
     public void overwriteWithContext(final ConfigContext context) {
         /* If a default context is being used to overwrite after this
          * context has been initialized, then we want to be careful to not
-         * overwrite values that have already been set with defaults. */
+         * overwrite values that have already been set by non-default contexts. */
         boolean isDefaultContext = context instanceof DefaultsConfigContext;
```

Comment:
Do we want this logic to apply for all subclasses of DefaultConfigContext? I don't know what would be right or wrong, but that would be the inflection point to decide between .equals() and instanceof.